pkg/controller/release/release.go

```diff
@@ -184,22 +192,43 @@ func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.E
 	// NOTE(negz): We don't currently check the identity type because at the
 	// time of writing there's only one valid value (Google App Creds), and
 	// that value is required.
-	if id := p.Spec.Identity; id != nil { //nolint:exhaustive
-		switch id.Source { //nolint:exhaustive
-		case xpv1.CredentialsSourceInjectedIdentity:
-			if err := c.gcpInjectorFn(ctx, rc, nil, gke.DefaultScopes...); err != nil {
-				return nil, errors.Wrap(err, errFailedToInjectGoogleCredentials)
-			}
-		default:
-			creds, err := c.gcpExtractorFn(ctx, id.Source, c.client, id.CommonCredentialSelectors)
-			if err != nil {
-				return nil, errors.Wrap(err, errFailedToExtractGoogleCredentials)
+	if id := p.Spec.Identity; id != nil {
+		switch id.Type {
+		case helmv1beta1.IdentityTypeGoogleApplicationCredentials:
+			switch id.Source { //nolint:exhaustive
+			case xpv1.CredentialsSourceInjectedIdentity:
+				if err := c.gcpInjectorFn(ctx, rc, nil, gke.DefaultScopes...); err != nil {
+					return nil, errors.Wrap(err, errFailedToInjectGoogleCredentials)
+				}
+			default:
+				creds, err := c.gcpExtractorFn(ctx, id.Source, c.client, id.CommonCredentialSelectors)
+				if err != nil {
+					return nil, errors.Wrap(err, errFailedToExtractGoogleCredentials)
+				}
+
+				if err := c.gcpInjectorFn(ctx, rc, creds, gke.DefaultScopes...); err != nil {
+					return nil, errors.Wrap(err, errFailedToInjectGoogleCredentials)
+				}
 			}
-
-			if err := c.gcpInjectorFn(ctx, rc, creds, gke.DefaultScopes...); err != nil {
-				return nil, errors.Wrap(err, errFailedToInjectGoogleCredentials)
+		case helmv1beta1.IdentityTypeAzurePrincipalCredentials:
+			switch id.Source { //nolint:exhaustive
+			case xpv1.CredentialsSourceInjectedIdentity:
+				if err := c.azureInjectorFn(ctx, rc, nil); err != nil {
```

Comment:
I don't think we would be supporting injectedIdentity currently. 
Could you double check how we would behave, e.g. when we pass credentials argument as `nil` and consider erroring out as "not supported" for now.